web/src/custom_profile_fields_ui.ts

```diff
@@ -145,15 +145,29 @@ export function initialize_custom_user_type_fields(
             field.type === field_types.USER.id &&
             (field_value_raw !== undefined || is_target_element_editable)
         ) {
-            const $pill_container = $(element_id)
-                .find(
-                    `.custom_user_field[data-field-id="${CSS.escape(`${field.id}`)}"] .pill-container`,
-                )
-                .expectOne();
+            let $pill_container = $(element_id).find(
+                `.custom_user_field[data-field-id="${CSS.escape(`${field.id}`)}"] .pill-container`,
+            );
+
+            // If there is no pill-container, fall back to view-only-pill.
+            if ($pill_container.length === 0) {
+                $pill_container = $(element_id).find(
+                    `.custom_user_field[data-field-id="${CSS.escape(`${field.id}`)}"] .view-only-pill`,
+                );
```

Comment:
Can we just directly use `user_display_only_pill` template for the display only pills? I can see in HTML content of the modal view that a contenteditable input is still present which is hidden.